custom_components/evnex/__init__.py

```diff
@@ -302,3 +306,64 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def _async_migrate_entries(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> bool:
+    """Migrate old entry."""
+    entity_registry = er.async_get(hass)
+
+    @callback
+    def update_unique_id(entry: er.RegistryEntry) -> dict[str, str] | None:
+        replacements = {
+            Platform.SENSOR.value: {
+                "org_wide_power_usage_today": "_org_wide_power_usage_today",
+                "org_wide_charger_sessions_today": "_org_wide_charger_sessions_today",
+                "org_tier": "_org_tier",
+                "charger_network_status": "_charger_network_status",
+                "session_energy": "_session_energy",
+                "session_cost": "_session_cost",
+                "session_time": "_session_time",
+                "session_start_time": "_session_start_time",
+                "charger_session_history": "_charger_session_history",
+                "_1_connector_current": "_1_connector_current_l1",
+                "_1_connector_voltage": "_1_connector_voltage_l1",
+            },
+            Platform.SWITCH.value: {
+                "charger_charge_now_switch": "_charger_charge_now",
+                "_1_connector_1_availability_switch": "_1_connector_1_availability",
+            },
+            Platform.BUTTON.value: {
+                "charger_stop_session": "_charger_stop_session",
+            },
+        }
+        uuid_part = entry.unique_id[:36]  # UUID is always 36 chars with dashes
+        remainder = entry.unique_id[36:]
+        if (key := remainder) in replacements.get(entry.domain, []):
+            new_unique_id = entry.unique_id.replace(
+                f"{uuid_part}{key}", f"{uuid_part}{replacements[entry.domain][key]}"
+            )
+            _LOGGER.debug(
+                "Migrating entity '%s' unique_id from '%s' to '%s'",
+                entry.entity_id,
+                entry.unique_id,
+                new_unique_id,
+            )
+            if existing_entity_id := entity_registry.async_get_entity_id(
+                entry.domain, entry.platform, new_unique_id
+            ):
+                _LOGGER.debug(
+                    "Cannot migrate to unique_id '%s', already exists for '%s'",
+                    new_unique_id,
+                    existing_entity_id,
+                )
+                return None
+            return {
+                "new_unique_id": new_unique_id,
+            }
+        return None
+
+    await er.async_migrate_entries(hass, config_entry.entry_id, update_unique_id)
```

Comment:
Nice, I hadn't come across HA entity migrations.

Should the migration logic be pulled out of `__init__py` though?

Reply:
A better approach would be to include within a bump of the entry version so it only occurs once, see https://github.com/home-assistant/core/blob/922eec09098e0462b64bea91210a674512b8530a/homeassistant/components/axis/__init__.py#L42

Reply:
Ahh yes, good point. Will pop that in.

Reply:
done